### PR TITLE
issue 511.7.3: restore single-document optimization, implement call to update data-store options, add IResearchView upgrade step

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

File: arangod/IResearch/IResearchFeature.cpp [hunk 1]
```diff
@@ -156,6 +159,180 @@ size_t computeThreadPoolSize(size_t threads, size_t threadsLimit) {
...
+             "0 to version 1";
+
+      return false;  // view drom failure
+    }
+
+    // .........................................................................
+    // non-recoverable state below here
+    // .........................................................................
+
+    // non-version 0 IResearchView implementations no longer drom from vocbase
```
Comment (graetzer) [inline arangod/IResearch/IResearchFeature.cpp]:
```suggestion
    // non-version 0 IResearchView implementations no longer from from vocbase
```

File: arangod/IResearch/IResearchFeature.cpp [hunk 2]
```diff
@@ -277,6 +454,44 @@ void registerRecoveryHelper() {
...
+
+  // move IResearch data-store from IResearchView to IResearchLink
+  {
+    arangodb::methods::Upgrade::Task task;
+
+    task.name = "IResearhView version 0->1";
+    task.description =
+        "move IResearch data-store from IResearchView to IResearchLink";
+    task.systemFlag = arangodb::methods::Upgrade::Flags::DATABASE_ALL;
+    task.clusterFlags = arangodb::methods::Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL  // any 1 single coordinator
```
Comment (graetzer) [inline arangod/IResearch/IResearchFeature.cpp]: I don't get it, is it necessary to really drop the view on the coordinator ?

File: arangod/IResearch/IResearchFeature.cpp (hunk 1 above)
Comment (vasiliy-arangodb) [reply arangod/IResearch/IResearchFeature.cpp]: done

File: arangod/IResearch/IResearchFeature.cpp (hunk 2 above)
Comment (vasiliy-arangodb) [reply arangod/IResearch/IResearchFeature.cpp]: you mean Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL?
this is to ensure that:
* the view definition in the agency is updated
* the links get forcefully recreated on all the dbservers

**Pull request merged**